### PR TITLE
Make http request only when spans is not empty

### PR DESCRIPTION
**Pull request by TokiTori**

### CODE REVIEW

**Comment by tsloughter** [comment]
@TokiTori to be clear, you are seeing this happen I take it? If it is an empty span list then this shouldn't be called in the first place so shouldn't need a check for an empty list. But you are seeing this happen?

I suppose it makes sense to also check if after attempting to encode it drops all the spans and ends up with an empty list, so I think it makes sense to merge this.

File: src/opentelemetry_zipkin.erl [hunk 1]
```diff
@@ -37,17 +37,23 @@ export(Tab, Resource, #state{address=Address,
...
-            ok;
-        {ok, {{_, Code, _}, _, Message}} ->
-            ?LOG_INFO("error response from service exported to status=~p ~p",
-                      [Code, Message]),
-            error;
-        {error, Reason} ->
-            ?LOG_INFO("client error exporting spans ~p", [Reason]),
-            error
+    if
+    ZSpans == [] ->
```
Comment (tsloughter) [inline src/opentelemetry_zipkin.erl]: Pls indent here, or even switch to use of a `case`, to match the style of other opentelemetry erlang code.

**Comment by TokiTori** [comment]
> @TokiTori to be clear, you are seeing this happen I take it? If it is an empty span list then this shouldn't be called in the first place so shouldn't need a check for an empty list. But you are seeing this happen?
> 
> I suppose it makes sense to also check if after attempting to encode it drops all the spans and ends up with an empty list, so I think it makes sense to merge this.

As I understand it, the `export` function is called from `opentelemetry_api`. After executing `export` ets table clear by another code. I checked it: each iteration of `export` has another set of spans in ets or has not spans even if last export fail.

I have no reason execute http request with empty spans - that is the purpose of this PR.

File: src/opentelemetry_zipkin.erl (hunk 1 above)
Comment (TokiTori) [reply src/opentelemetry_zipkin.erl]: Ok. I'll fix it.

Comment (TokiTori) [reply src/opentelemetry_zipkin.erl]: I have overwritten my previous commit. Please check it.

**Comment by TokiTori** [comment]
After exporting ets table is deleted and recreated: https://github.com/open-telemetry/opentelemetry-erlang/blob/d33a383f41a4d6c4dee4883ce5eb54e2230a7c74/apps/opentelemetry/src/otel_batch_processor.erl#L274

**Comment by tsloughter** [comment]
Thanks, yea, I was thinking of OpenCensus, which actually grabbed all the spans in the table before passing to the exporter so it could do the `case Spans of [] -> skip` itself. We will likely want to change the batch processor to not call the exporter if the table is empty (just need an `ets:info(T, size)` or whatever check)  but for now this is good enough.

**Pull request merged**